contracts/OffsetHelper.sol

```diff
@@ -57,66 +56,102 @@ import "./interfaces/IToucanContractRegistry.sol";
  */
 contract OffsetHelper is OffsetHelperStorage {
     using SafeERC20 for IERC20;
-
-    /**
-     * @notice Contract constructor. Should specify arrays of ERC20 symbols and
-     * addresses that can used by the contract.
-     *
-     * @dev See `isEligible()` for a list of tokens that can be used in the
-     * contract. These can be modified after deployment by the contract owner
-     * using `setEligibleTokenAddress()` and `deleteEligibleTokenAddress()`.
-     *
-     * @param _eligibleTokenSymbols A list of token symbols.
-     * @param _eligibleTokenAddresses A list of token addresses corresponding
-     * to the provided token symbols.
-     */
-    constructor(
-        string[] memory _eligibleTokenSymbols,
-        address[] memory _eligibleTokenAddresses
-    ) {
-        uint256 i = 0;
-        uint256 eligibleTokenSymbolsLen = _eligibleTokenSymbols.length;
-        while (i < eligibleTokenSymbolsLen) {
-            eligibleTokenAddresses[
-                _eligibleTokenSymbols[i]
-            ] = _eligibleTokenAddresses[i];
-            i += 1;
-        }
-    }
+    // Chain ID of Celo mainnet
+    uint256 private constant CELO_MAINNET_CHAINID = 42220;
```

Comment:
Is there any reason to hardcode the chain id as part of the contract? We should try to keep contracts chain-agnostic.

Reply:
If this is used to workaround an issue specific to Celo, it'd be best to add a comment for why we need this.

Reply:
This will still be problematic for Celo forks.

Reply:
So, Celo allows to pay the gas fees with tokens other than the native token, which is why the swapETH functions don't work on Celo.  
Maybe this might change once they upgrade to the OP stack...